install.sh

```diff
@@ -1019,9 +1019,9 @@ then
   then
     plural="s"
   fi
-  printf "- Run these commands in your terminal to add the non-default Git remote%s for %s:" "${plural}" "${non_default_repos}"
-  printf "    echo '# Set PATH, MANPATH, etc., for Homebrew.' >> %s" "${shell_profile}"
-  printf "    echo '%s' >> %s\n" "${additional_shellenv_commands[@]}" "${shell_profile}"
+  echo "- Run these commands in your terminal to add the non-default Git remote${plural} for ${non_default_repos}:"
```

Comment:
Why replace it with an echo only for the first line?

Reply:
I refer to the code of the last macOS version (Monterey):  https://github.com/Homebrew/install/commit/cc459e4c030b5a6a4aa42e0ffdd13e54cedbcddb

Reply:
@dirname Sorry, I don't understand the justification here. What's your `brew config` output?

Reply:
> @dirname Sorry, I don't understand the justification here. What's your `brew config` output?

I set `HOMEBREW_BREW_GIT_REMOTE`, `HOMEBREW_CORE_GIT_REMOTE` to use the mirror to install brew, and then an error occurred as described in the picture `/bin/bash: line 1021: printf: - : invalid option`. This may be caused by `/bin/bash -c`, so I refer to the code that the previous version can install.

https://github.com/Homebrew/install/blob/684032061383ab14862240aaff4273cde4e1847c/install.sh#L1022

Similar error: 
`/bin/bash -c 'printf "hello"'` will print `hello` normally, but `/bin/bash -c 'printf "-hello"'` will cause same error.

![image](https://user-images.githubusercontent.com/18101258/202192776-9ad3a7c8-0f0a-4234-a843-2d6a1f0b3a5d.png)

Reply:
I'd suggest we use `printf -- "-hello"` instead which will fix this more simply and not require changing as much.

Reply:
> I'd suggest we use `printf -- "-hello"`

This looks like the safest solution. Alternative ones are to use `echo` (as was suggested originally), replace `-` with `\055`, or pass the message as a parameter (`printf "%s\n" "- Run these"`)

I think it's a perfect time for a reminder that in Bash double quotes are used to prevent word splitting, not to indicate a string (as in other programming languages). So, things like `var=""` are meaningless and the proper way to write the same in Bash is `var=`.